### ISSUE TITLE
Installation crashes in LogisticLossSimple

### ISSUE THREAD

**Issue by ido** [probml/pmtk3#9]
_From [mathis...@gmail.com](https://code.google.com/u/104684413862879970216/) on November 23, 2010 14:44:22_

What steps will reproduce the problem? 1. Follow the installation instruction: http://code.google.com/p/pmtk3/wiki/installation 2. Macintosh; matlab 7.9.0.529 ( R2009b )
3. testPmtk3 crashes. What is the expected output? What do you see instead? I expected the demos to run. Instead I get:
??? Undefined function or variable 'p'.

Error in ==> LogisticLossSimple at 16
Xw = X .\* repmat(rowvec(weights), 1, p);

Sure enough, LogisticLossSimple refers to an undefined variable 'p'. What version / revision of the product are you using? On what operating system? pmtk3-24October2010.zip
MacOS 10.6.5 Please provide any additional information below. Here's the command history in matlab:

> > initPmtk3
> > initializing pmtk3
> > downloading GGM-GWishart.............done
> > downloading SPAMS-1.02...............failed
> > downloading bpca.....................done
> > downloading dpmixturesTeh07..........done
> > downloading ekfukf1.2................done
> > downloading export_fig...............done
> > downloading exportfig................done
> > downloading fastICA-2.5..............done
> > downloading fastfit..................done
> > downloading gaimc1.0-graphAlgo.......done
> > downloading glmnet-matlab............done
> > downloading gpml-matlab..............done
> > downloading graphViz4Matlab..........done
> > downloading l1ls.....................done
> > downloading lars.....................done
> > downloading libdai-0.2.6.............done
> > downloading liblinear-1.51...........done
> > downloading libsvm-mat-2.9.1.........done
> > downloading lightspeed2.3............done
> > downloading markSchmidt-21june2010...done
> > downloading matbugs..................done
> > downloading matlabRlink..............done
> > downloading mcmcdiag.................failed
> > downloading mplp-1.0.................done
> > downloading netlab3.3................done
> > downloading onlineEM.................done
> > downloading randraw..................done
> > downloading sparseBayes2.0...........done
> > downloading svmLightWindows..........done
> > downloading vblinlogreg..............done
> > welcome to pmtk3
> > testPmtk3
> > Checking for matlabTools.........PASSED
> > Checking initialization..........PASSED
> > Checking pmtkSupport packages....PASSED
> > Checking pmtkData................downloading http://pmtkdata.googlecode.com/svn/trunk/crabs/crabs.zip to /Users/don/Documents/MATLAB/MurphySimulator/pmtk3/data/crabs.zip
> > unzipped and loaded crabs.mat
> > PASSED
> > Checking for graphviz............Please install or upgrade graphViz
> > Please install or upgrade graphViz
> > Please install or upgrade graphViz
> > Please install or upgrade graphViz
> > Please install or upgrade graphViz
> > FAILED
> > Testing selected code............

nll =
    6.9315
??? Undefined function or variable 'p'.

Error in ==> LogisticLossSimple at 16
Xw = X .\* repmat(rowvec(weights), 1, p);

Error in ==> logregFit>@(w)LogisticLossSimple(w,X,y,weights) at 101
    loss = @(w) LogisticLossSimple(w, X, y, weights);

Error in ==> penalizedL2 at 13
    [nll,g] = gradFunc(w,varargin{:});

Error in ==> logregFit>@(w)penalizedL2(w,loss,lambdaVec(:)) at 133
    penloss = @(w)penalizedL2(w, loss, lambdaVec(:));

Error in ==> minFunc at 303
    [f,g] = funObj(x,varargin{:});

Error in ==> logregFit at 135
    [w, opt.finalObj, opt.exitflag, opt.output] = ...

Error in ==> logregL2FitTest at 19
model1 = logregFit(X, y01, 'lambda', lambda, 'preproc', pp, 'fitOptions', opts);

Error in ==> testPmtk3 at 66
logregL2FitTest;

_Original issue: http://code.google.com/p/pmtk3/issues/detail?id=9_
